core/src/main/java/org/infinispan/configuration/cache/TransactionConfigurationBuilder.java

```diff
@@ -327,6 +332,18 @@ public void validate() {
       if (attributes.attribute(TRANSACTION_MODE).get() == TransactionMode.TRANSACTIONAL && !cacheMode.isSynchronous()) {
          throw log.unsupportedAsyncCacheMode(cacheMode);
       }
+      if (attributes.attribute(TRANSACTION_MODE).get() == TransactionMode.TRANSACTIONAL &&
+            attributes.attribute(LOCKING_MODE).get() == LockingMode.OPTIMISTIC &&
+            !attributes.attribute(USE_EAGER_LOCKING).get() && //this is pessimistic transaction
+            getBuilder().locking().isolationLevel() == IsolationLevel.REPEATABLE_READ &&
+            cacheMode != CacheMode.INVALIDATION_SYNC && //write-skew only for REPL, DIST and LOCAL (async is not supported in tx mode)
+            !getBuilder().invocationBatching().isEnabled()) { //batching can live without WSC
+         getBuilder().locking().writeSkewCheck(true);
+         getBuilder().versioning().enable().scheme(VersioningScheme.SIMPLE);
```

Comment:
Put a warning if the write-skew is disabled. We should set write-skew as `true` by default in `LockingConfiguration`. Then, we should put a warning if the locking mode was set to pessimistic, but write skew was not disabled.
The code should ignore the actual value of write-skew in configuration.